src/js/modules/FormFeedback.js

```diff
@@ -0,0 +1,231 @@
+import React, { Component } from 'react';
+import { defineMessages, injectIntl } from 'react-intl';
+import { postFeedback } from 'js/helpers/fetchData';
+import { logEvent } from 'js/helpers/googleAnalytics';
+import SectionHeaderSerif from 'js/modules/SectionHeaderSerif';
+
+import emojiDisappointed from 'images/emojis/disappointed.png';
+import emojiSad from 'images/emojis/sad.png';
+import emojiNeutral from 'images/emojis/neutral.png';
+import emojiSlightlySmiling from 'images/emojis/slightly_smiling.png';
+import emojiGrinning from 'images/emojis/grinning.png';
+
+const i18nMessages = defineMessages({
+  step1title: {
+    id: 'FormFeedback.step1.title',
+    defaultMessage: 'How satisfied are you with this page?',
+  },
+  step2titlea: {
+    id: 'FormFeedback.step2.title.a',
+    defaultMessage: 'Thank you!',
+  },
+  step2titleb: {
+    id: 'FormFeedback.step2.title.b',
+    defaultMessage: 'How can we make this page better?',
+  },
+  step3title: {
+    id: 'FormFeedback.step3.title',
+    defaultMessage: 'Thank you for sharing feedback!',
+  },
+  step3button: {
+    id: 'FormFeedback.step3.button',
+    defaultMessage: 'Give more feedback',
+  },
+  emojiDisappointed: {
+    id: 'FormFeedback.emojiDisappointed.altText',
+    defaultMessage: 'Disappointed',
+  },
+  emojiSad: {
+    id: 'FormFeedback.emojiSad.altText',
+    defaultMessage: 'Sad',
+  },
+  emojiNeutral: {
+    id: 'FormFeedback.emojiNeutral.altText',
+    defaultMessage: 'Neutral',
+  },
+  emojiSlightlySmiling: {
+    id: 'FormFeedback.emojiSlightlySmiling.altText',
+    defaultMessage: 'Slightly Smiling',
+  },
+  emojiGrinning: {
+    id: 'FormFeedback.emojiGrinning.altText',
+    defaultMessage: 'Grinning',
+  },
+});
+
+
+class FormFeedback extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      step: 1,
+      loading: false,
+      emoji: null,
+      feedback: null,
+      error: null
+    };
+  }
+
+  logEvent(action, label) {
+    logEvent({
+     category: 'FORM__site-feedback',
+     action: `${action}__step-${this.state.step}`,
+     label: label
+    });
+  }
+
+  handleEmojiClick = (e) => {
+    this.logEvent('emoji-click', JSON.stringify({
+        'emoji':e.currentTarget.value
+      }) );
+
+    this.setState({
+      step: 2,
+      emoji: e.currentTarget.value
+    });
+  }
+
+  handleTextAreaChange = (e) => {
+    this.setState({
+      feedback: e.currentTarget.value
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    this.logEvent('send-feedback-click', JSON.stringify({
```

Comment:
Why do we have to stringify this object as we pass it as an argument? Would it be better for that stringify-ication to happen inside the function if we're going to have to do it all the time when its invoked?

Reply:
Added a logFormEvent which JSON.stringifies in one place.